src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/DefaultGetCaseViewOperation.java

```diff
@@ -52,54 +64,61 @@ public DefaultGetCaseViewOperation(@Qualifier(CreatorGetCaseOperation.QUALIFIER)
-        if (!uidService.validateUID(caseReference)) {
-            throw new BadRequestException("Case reference " + caseReference + " is not valid");
-        }
+        validateCaseReference(caseReference);
 
         final CaseType caseType = getCaseType(jurisdictionId, caseTypeId);
-        final CaseDetails caseDetails =
-            getCaseOperation.execute(jurisdictionId, caseTypeId, caseReference)
-                            .orElseThrow(() -> new ResourceNotFoundException(
-                                String.format(RESOURCE_NOT_FOUND, jurisdictionId, caseTypeId, caseReference)));
+        final CaseDetails caseDetails = getCaseDetails(jurisdictionId, caseTypeId, caseReference);
 
         final List<AuditEvent> events = listEventsOperation.execute(caseDetails);
-        final CaseTabCollection caseTabCollection = uiDefinitionRepository.getCaseTabCollection(caseDetails.getCaseTypeId());
+        final CaseTabCollection caseTabCollection = getCaseTabCollection(caseDetails.getCaseTypeId());
 
         return merge(caseDetails, events, caseType, caseTabCollection);
     }
 
+    @Override
+    public CaseHistoryView execute(String jurisdictionId, String caseTypeId, String caseReference, Long eventId) {
+        validateCaseReference(caseReference);
+
+        CaseType caseType = getCaseType(jurisdictionId, caseTypeId);
+        CaseDetails caseDetails = getCaseDetails(jurisdictionId, caseTypeId, caseReference);
+
+        AuditEvent event = listEventsOperation.execute(jurisdictionId, caseTypeId, eventId);
+
+        return merge(caseDetails, event, caseType);
+    }
+
+    private void validateCaseReference(String caseReference) {
+        if (!uidService.validateUID(caseReference)) {
+            throw new BadRequestException("Case reference " + caseReference + " is not valid");
+        }
+    }
+
     private CaseType getCaseType(String jurisdictionId, String caseTypeId) {
         return caseTypeService.getCaseTypeForJurisdiction(caseTypeId, jurisdictionId);
     }
 
+    private CaseDetails getCaseDetails(String jurisdictionId, String caseTypeId, String caseReference) {
+        return getCaseOperation.execute(jurisdictionId, caseTypeId, caseReference)
+            .orElseThrow(() -> new ResourceNotFoundException(
+                String.format(RESOURCE_NOT_FOUND, jurisdictionId, caseTypeId, caseReference)));
+    }
+
+    private CaseTabCollection getCaseTabCollection(String caseTypeId) {
+        return uiDefinitionRepository.getCaseTabCollection(caseTypeId);
+    }
+
     private CaseView merge(CaseDetails caseDetails, List<AuditEvent> events, CaseType caseType, CaseTabCollection caseTabCollection) {
         CaseView caseView = new CaseView();
         caseView.setCaseId(caseDetails.getReference().toString());
-        caseView.setChannels(caseTabCollection.getChannels().toArray(new String[caseTabCollection.getChannels().size()]));
+        caseView.setChannels(caseTabCollection.getChannels().toArray(new String[0]));
```

Comment:
👍

Reply:
event though, I think in the future we should stop using arrays and move to List :)

Reply:
Yeah tried to refactor the code to list initially but there were many tests affected so thought I should not have so many code changes in my first PR ;-)